### PR TITLE
Fixed #28571 -- Added a prompt to bypass password validation in createsuperuser.

### PR DESCRIPTION
**Pull request by orf**
[Ticket](https://code.djangoproject.com/ticket/28571)

### CODE REVIEW

**Comment by orf** [comment]
Tests are failing, not sure why? Seems to be a runner issue rather than the code as far as I can see?

**Comment by timgraham** [comment]
The build is hanging so I aborted it. Possibly this prompt is appearing in some other test? I couldn't reproduce it locally yet.

**Comment by orf** [comment]
Ahh yes, I can see where. I will fix it tonight.

These tests are quite magical, especially with the fake input decorator. It would be nice to be able to supply a hard list of expected prompts and the inputs, which will throw an error if not matched. 

Currently if you get it wrong the tests will just spin forever in a while loop inside `createsuperuser`

**Comment by orf** [comment]
Found a fix, the prompt had `password` in it which would (depending on dictionary iteration order) trigger the `password` mock input rather than `bypass`

**Comment by timgraham** [comment]
Oh, I did make that addition of 'password' in the prompt. I thought the problem only appeared after my edit but couldn't think how it could be related. Perhaps a refactor of the testing approach is needed as I'd like not to be limited to writing our prompts based on it (plus, that's annoying to debug if it happens in the future). :-)

**Comment by orf** [comment]
I'm happy to do that, it's only used in a few places right now. Shall I do it as a separate MR? It would need changing in all places it's used to fix it, I think.

**Comment by timgraham** [comment]
Yes, we would merge that separately before this.

**Comment by timgraham** [comment]
Tom, I added a test refactor commit to this PR. Could you review it?

File: tests/auth_tests/test_management.py [hunk 1]
```diff
@@ -41,15 +47,24 @@ def getpass(prompt=b'Password: ', stream=None):
                     return inputs['password']
 
             def mock_input(prompt):
+                # This must be interpolated at runtime rather than during
+                # module loading.
+                MOCK_INPUT_KEY_TO_PROMPTS['username'][1] %= {
+                    'default_username': get_default_username(),
```
Comment (orf) [inline tests/auth_tests/test_management.py]: I had no idea you could do `%=`, interesting! This does feel a bit too special cased though, couldn't you add a lambda function to the `MOCK_INPUT_KEY_TO_PROMPTS`, i.e: `['Username: ', lambda: "Username (leave blank to use '%s'): " % get_default_username()]`, then call it if needed below?

It's not that important though.

**Comment by orf** [comment]
Thank you! It's been on my to-do list but I've got limited free time right now.

It looks fine, I left one small comment but it seems much less fragile than the previous approach.

File: tests/auth_tests/test_management.py (hunk 1 above)
Comment (timgraham) [reply tests/auth_tests/test_management.py]: Good suggestion, updated.

**Pull request merged**